src/Process/LayeredMaterialNodeProcessing.js

```diff
@@ -237,6 +237,9 @@ export function updateLayeredMaterialNodeElevation(context, layer, node, parent)
 
 export function removeLayeredMaterialNodeLayer(layerId) {
     return function removeLayeredMaterialNodeLayer(node) {
+        if (node.geometry) {
+            node.geometry.dispose();
+        }
```

Comment:
I agree with you that `The geometry of each tile should not depend on layers attached to the tile.` 
However, I thought that `node.geometry` represented the own geometry of the `RasterLayer` if it was not attached, but now I'm not sure of it. Can a`RasterLayer` have its own geometry or is it always attached to a GeometryLayer? Is `node.geometry` always the geometry of the layer it is attached too?

Reply:
`RasterLayers` are always attached to another `GeometryLayer` - which most of the times is a [`TiledGeometryLayer`](https://github.com/iTowns/itowns/blob/master/src/Layer/TiledGeometryLayer.js). This `TiledGeometryLayer` displays the globe -or planar- tiles with a default material (the blue color we see if no `ColorLayer` is attached). When we add a `RasterLayer` : 
- if it is a `ColorLayer`, the image is used as a texture for the tile material,
- if it is an `ElevationLayer`, the `z` coordinate of each tile vertex is translated by the correct elevation.

When removing a `RasterLayer`, we take the tiles from the minimum zoom level, and we traverse all their children which are passed to `removeLayeredMaterialNodeLayer` parameter. Therefore, the `node` in this case is always a tile from the `TiledGeometryLayer` to which the `RasterLayer` is attached.

To conclude and answer your question : 
- a `RasterLayer` does not have its own geometry and is always attached to a `GeometryLayer`.
- `node.geometry` is always the geometry of the layer the `RasterLayer` is attached to.

I hope this is clear enough, don't hesitate if it needs clarification.

_PS : each tile from a `TiledGeometryLayer` actually has five children : the four tiles matching the subsequent zoom level subdivision, and the tile's OBB. Perhap this could be improved, having the OBB as a property and not a children._

Reply:
Ok thanks for the explaination, it is clearer now! I removed this line and I added your explanation in the documentation.

Can you open issues about the problem you mention and about your proposition for OBB management please?

Reply:
#1983 opened on this subject.